### PR TITLE
py-jupytext: add new package

### PR DESCRIPTION
**Pull request by vvolkl**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-jupytext/package.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    version('1.13.0', sha256='fb220af65d2bd32d01c779b0e935c4c2b71e3f5f2f01bf1bab10d5f23fe121d4')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-jupyter-packaging', type=('build', 'run'))
+    depends_on('py-jupyterlab', type=('build', 'run'))
+    depends_on('py-nbformat', type=('build', 'run'))
+    depends_on('py-pyyaml', type=('build', 'run'))
+    depends_on('py-toml', type=('build', 'run'))
+    depends_on('py-mdit-py-plugins', type=('build', 'run'))
+    depends_on('py-markdown-it-py', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jupytext/package.py]:
```suggestion
    depends_on('py-markdown-it-py@1.0:1', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-jupytext/package.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    homepage = "https://github.com/mwouts/jupytext/"
+    git      = "https://github.com/mwouts/jupytext/"
+    pypi     = "jupytext/jupytext-1.13.0.tar.gz"
+
+    maintainers = ['vvolkl']
+
+    version('1.13.0', sha256='fb220af65d2bd32d01c779b0e935c4c2b71e3f5f2f01bf1bab10d5f23fe121d4')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-jupyter-packaging', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jupytext/package.py]: It doesn't seem like this dep is used unless we are building the jupyterlab extensions?

File: var/spack/repos/builtin/packages/py-markdown-it-py/package.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    """Markdown parser, done right.
+       100% CommonMark support, extensions, syntax plugins & high speed """
+
+    homepage = "https://github.com/executablebooks/markdown-it-py"
+    git      = "https://github.com/executablebooks/markdown-it-py"
+    pypi     = "markdown-it-py/markdown-it-py-1.1.0.tar.gz"
+
+    version('1.1.0', sha256='36be6bb3ad987bfdb839f5ba78ddf094552ca38ccbd784ae4f74a4e1419fc6e3')
+
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-markdown-it-py/package.py:21]: The deps for this one are hidden in `setup.cfg`:
```
install_requires =
        attrs>=19,<22
        typing_extensions>=3.7.4;python_version<'3.8'
python_requires = ~=3.6
```

File: var/spack/repos/builtin/packages/py-mdit-py-plugins/package.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+class PyMditPyPlugins(PythonPackage):
+    """Collection of core plugins for markdown-it-py"""
+
+    homepage = "https://github.com/executablebooks/mdit-py-plugins/"
+    git      = "https://github.com/executablebooks/mdit-py-plugins/"
+    pypi     = "mdit-py-plugins/mdit-py-plugins-0.2.8.tar.gz"
+
+    version('0.2.8', sha256='5991cef645502e80a5388ec4fc20885d2313d4871e8b8e320ca2de14ac0c015f')
+
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdit-py-plugins/package.py:19]: Same here:
```
install_requires =
        markdown-it-py~=1.0
python_requires = ~=3.6
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by vvolkl** [comment]
@adamjstewart Thanks for the comments! Updated now.

File: var/spack/repos/builtin/packages/py-markdown-it-py/package.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    homepage = "https://github.com/executablebooks/markdown-it-py"
+    git      = "https://github.com/executablebooks/markdown-it-py"
+    pypi     = "markdown-it-py/markdown-it-py-1.1.0.tar.gz"
+
+    version('1.1.0', sha256='36be6bb3ad987bfdb839f5ba78ddf094552ca38ccbd784ae4f74a4e1419fc6e3')
+
+    depends_on('python@3.6:',   type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('py-attrs@19:22', type=('build', 'run'))
+    depends_on('py-typing-extensions@3.7.4:', type=('build', 'run'), when='^python@:3.8')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-markdown-it-py/package.py]:
```suggestion
    depends_on('py-attrs@19:21', type=('build', 'run'))
    depends_on('py-typing-extensions@3.7.4:', type=('build', 'run'), when='^python@:3.7')
```
Spack version ranges are inclusive.

File: var/spack/repos/builtin/packages/py-markdown-it-py/package.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    """Markdown parser, done right.
+       100% CommonMark support, extensions, syntax plugins & high speed """
+
+    homepage = "https://github.com/executablebooks/markdown-it-py"
+    git      = "https://github.com/executablebooks/markdown-it-py"
+    pypi     = "markdown-it-py/markdown-it-py-1.1.0.tar.gz"
+
+    version('1.1.0', sha256='36be6bb3ad987bfdb839f5ba78ddf094552ca38ccbd784ae4f74a4e1419fc6e3')
+
+    depends_on('python@3.6:',   type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-markdown-it-py/package.py]:
```suggestion
    depends_on('python@3.6:3',   type=('build', 'run'))
```
`~=` is different than `>=`

File: var/spack/repos/builtin/packages/py-mdit-py-plugins/package.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    depends_on('py-markdown-it-py@1.0:', type=('build', 'run'))
+    depends_on('python@3.6:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdit-py-plugins/package.py]:
```suggestion
    depends_on('py-markdown-it-py@1.0:1', type=('build', 'run'))
    depends_on('python@3.6:3', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-markdown-it-py/package.py (hunk 6 above)
Comment (vvolkl) [reply var/spack/repos/builtin/packages/py-markdown-it-py/package.py]: Planning for python 4 already? That's futureproof :)

File: var/spack/repos/builtin/packages/py-jupytext/package.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    homepage = "https://github.com/mwouts/jupytext/"
+    git      = "https://github.com/mwouts/jupytext/"
+    pypi     = "jupytext/jupytext-1.13.0.tar.gz"
+
+    maintainers = ['vvolkl']
+
+    version('1.13.0', sha256='fb220af65d2bd32d01c779b0e935c4c2b71e3f5f2f01bf1bab10d5f23fe121d4')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-jupyterlab', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jupytext/package.py]: Is this dependency actually needed? I don't see it listed in the `setup.py`.

Comment (vvolkl) [reply var/spack/repos/builtin/packages/py-jupytext/package.py]: Not sure if it's actually needed, have it from here: https://github.com/mwouts/jupytext/blob/main/pyproject.toml#L2

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-jupytext/package.py]: Spack doesn't currently use that file, we run `python setup.py install` instead. Can you remove it and see if the build succeeds without downloading that dependency?

Comment (vvolkl) [reply var/spack/repos/builtin/packages/py-jupytext/package.py]: Works indeed without `py-jupyterlab`, so I removed it as a dependency.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**